core/contracts/Voting.sol

```diff
@@ -113,55 +122,66 @@ contract Voting is Testable {
     function requestPrice(bytes32 identifier, uint time) external returns (uint expectedTime) {
         // TODO: we may want to allow future price requests and/or add a delay so that the price has enough time to be
         // widely distributed and agreed upon before the vote. 
-        require(time < getCurrentTime());
+        uint blockTime = getCurrentTime();
+        require(time < blockTime);
 
+        uint currentRoundId = voteTiming.computeCurrentRoundId(blockTime);
         uint priceResolutionRound = _getPriceResolution(identifier, time).lastVotingRound;
-        uint secondsInWeek = 60*60*24*7;
 
         if (priceResolutionRound == 0) {
             // Price has never been requested.
+
+            // Must ensure the round is updated here so the requested price will be voted on in the next commit cycle.
+            // TODO: this may be an expensive operation - may make sense to have a public updateRound() method to handle
+            // extreme cases.
+            _updateRound(blockTime);
+
+            // Price requests always go in the next round, so add 1 to the computed current round.
+            uint nextRoundId = currentRoundId.add(1);
+
             // Add price request to the next round.
-            _addPriceRequestToRound(_calcVotingRound().add(1), PriceRequest(identifier, time));
-
-            // TODO: replace this with the end of the next round once phasing has been implemented.
-            return getCurrentTime().add(secondsInWeek.mul(2));
-        } else if (priceResolutionRound == currentRoundNumber) {
-            // Price is currently being resolved.
-            // TODO: replace this with the end of the current round once phasing has been implemented.
-            return getCurrentTime().add(secondsInWeek);
+            _addPriceRequestToRound(nextRoundId, PriceRequest(identifier, time));
+
+            // Estimate the end of next round and return the time.
+            return voteTiming.computeEstimatedRoundEndTime(nextRoundId);
+        } else if (priceResolutionRound >= currentRoundId) {
+            // Price is already slated to be resolved.
+            
+            return voteTiming.computeEstimatedRoundEndTime(priceResolutionRound);
         } else {
             // Price has been resolved.
```

Comment:
Good point, I just moved the call up for now because keeping it in the if statement seems like it would make the code more confusing as you mentioned earlier today.